### PR TITLE
Upgrade to controller runtime 0.7.0

### PR DESCRIPTION
**Pull request by stijndehaes**
I have upgraded this to controller runtime 0.7.0, I already provided the PR for feedback. Tests run succesful, but I still want to test the code in a pod on a cluster. Will update my comment when I have done this.

Ok I have it working on a cluster, also noticed it might be interesting to migrate to code to use v1 of  `admissionregistration.k8s.io`. If you guys are interested I'll do this in a next PR

### CODE REVIEW

**Comment by ritazh** [comment]
/hold 

Per community call discussion, this may need to wait a little bit depending on when controller-runtime upgrade work is done in Gatekeeper to minimize conflicts.

**Comment by stijndehaes** [comment]
@ritazh I understand, I can use my version of the code in the mean time so I am not in a hurry to get this merged :) I'll keep it up to date with master every once in a while

**Comment by adrianludwin** [comment]
We're trying to upgrade HNC to controller-runtime v0.7.0 too, but we're blocked on this PR going in. @ritazh , would you mind if we merged this change? As long as Gatekeeper is importing cert-controller at a pinned verion, it shouldn't affect your builds until you explicitly upgrade to the latest version, right?

File: pkg/rotator/rotator.go
Comment (adrianludwin) [inline pkg/rotator/rotator.go:542]: It looks like you've inlined all of this code into `addController` - what's the reason for that? `addController` is already fairly long. Either way, if you _do_ inline this into `addController`, shouldn't this code all be deleted, and not commented out? Also, can we make this part of a separate change, i.e. not part of the upgrade to controller-runtime 0.7.0?

Comment (stijndehaes) [reply pkg/rotator/rotator.go:542]: Forgot to properly clean this up will do that right now!
Indeed instead of inlining I can just pass a function. The reason for the change is simply that the function now a function instead of an interface, this is an upstream change in controller runtime

Comment (stijndehaes) [reply pkg/rotator/rotator.go]: Changed it please let me know if you think the current version is better

Comment (adrianludwin) [reply pkg/rotator/rotator.go]: lgtm, thanks!

**Comment by adrianludwin** [comment]
I've imported the latest version of @stijndehaes 's change into HNC as a part of also upgrading HNC to controller-runtime v0.7.0, and everything appears to be working (our basic tests have passed; the rest of the suite is still running). I'll update this if there are any problems but at this point I'm fairly confident that the change is good.

Thanks!

**Comment by sozercan** [comment]
We can unhold this now. https://github.com/open-policy-agent/gatekeeper/pull/1073 is out for upgrading CR to v0.7.0

**Review by maxsmythe — APPROVED** [review]
LGTM, waiting for LGTM from @ritazh or @sozercan before merging.

File: pkg/rotator/rotator.go
Comment (ritazh) [inline pkg/rotator/rotator.go:543]: why is `object.GetNamespace()` called here?

Comment (maxsmythe) [reply pkg/rotator/rotator.go:543]: ah, good call, this line looks like it doesn't do anything and should probably be removed.

File: pkg/rotator/rotator_suite_test.go
Comment (shomron) [inline pkg/rotator/rotator_suite_test.go:36]: Would it be cleaner to pass a context to this function instead of a creating it here?

File: pkg/rotator/rotator_test.go
Comment (shomron) [inline pkg/rotator/rotator_test.go:222]: Should we `defer cancelFunc()` in case there is an early exit? Same question for other tests.

File: pkg/rotator/rotator_suite_test.go
Comment (stijndehaes) [reply pkg/rotator/rotator_suite_test.go:36]: Looking in some more detail there is already a context created in the test func that is already used. So I might just pass that to this func. 
WDYT?

Comment (maxsmythe) [reply pkg/rotator/rotator_suite_test.go:36]: My guess is that Oren would be happy with that :)

Comment (shomron) [reply pkg/rotator/rotator_suite_test.go:36]: Super happy :)

File: pkg/rotator/rotator_test.go
Comment (stijndehaes) [reply pkg/rotator/rotator_test.go]: So for now I have created a context that is cancellabe in the root of the test and immeadiately after I call defer cancelFunc()

File: pkg/rotator/rotator_test.go
Comment (shomron) [inline pkg/rotator/rotator_test.go:183]: You'll need to call `cancelFunc` right before `wg.Wait()` - it is waiting for the manager to complete and the manager is waiting for the context to close.
This will not interfere with the `defer` (which still handles early exit) - as It is safe to call `cancelFunc` multiple times.

This is true for the other test as well.

**Review by shomron — CHANGES_REQUESTED** [review]

File: pkg/rotator/rotator_test.go
Comment (shomron) [reply pkg/rotator/rotator_test.go:183]: @stijndehaes thanks for working on this!  Do you have a sense on when you'd be able to address the above? We have another change that depends on this one and we'd love to see them all get merged :)

Comment (stijndehaes) [reply pkg/rotator/rotator_test.go:183]: Sorry missed the previous communication :) Fixed it now

Comment (shomron) [reply pkg/rotator/rotator_test.go:183]: Thank you!

**Review by sozercan — APPROVED** [review]
LGTM

**Comment by maxsmythe** [comment]
@shomron LGTY?

**Review by shomron — APPROVED** [review]
🚀

**Pull request merged**
